content/js1/sprints/2/prep/index.md

```diff
@@ -8,4 +8,406 @@ backlog= 'Module-JS1'
+
+The following variables represent the dimensions of a rectangle:
+
+```js
+height; // could have a value like 10
+width; // could have a value like 30
+```
+
+Write expressions for the following in terms of the variables above:
+a) area of the rectangle
+b) perimeter of the rectangle
+
+{{</note>}}
+
+{{<note type="exercise" title="Exercise 1.2">}}
+
+The following variable represents the cost of something in pence:
+
+```js
+price; // could have a value like "130p"
+```
+
+Write an expression for the price in pounds. The price in pounds should start with a "£" symbol. The price should be to 2 decimal places.
+
+{{</note>}}
+
+### ♻️ Wrapping up functionality
+
+Now we want to wrap up the expression we’ve created and reuse it to convert any number into a percentage.
+
+We can write a function declaration to reuse some code. A function declaration looks like this:
+
+```js {linenos=table,linenostart=1}
+function convertToPercentage() {}
+```
+
+The `function` declaration consists of the following syntactic elements:
+
+- `function` keyword, denotes the beginning of a function declaration
+- `convertToPercentage` - this is the name of the function
+- `{}` - the body of the function is written inside the curly braces
+
+We can wrap the functionality we want to reuse inside the curly braces by writing:
+
+```js {linenos=table,linenostart=1}
+function convertToPercentage() {
+  const num = 0.5;
+  const percentage = `${num * 100}%`;
+}
+```
+
+### 🔎 Checking the output
+
+Earlier we saw how to store a function's output in a variable.
+
+```js
+const result = Math.round(10.3);
+```
+
+In the code above, we call `Math.round` which takes the input `10.3` and then outputs the rounded number `10`. So `result` stores a value of `10`.
+
+Now we want to call the function `convertToPercentage` and store the output in a variable.
+We can store the return value in a variable in exactly the same way:
+
+```js {linenos=table,hl_lines=["5"],linenostart=1}
+function convertToPercentage() {
+  const num = 0.5;
+  const percentage = `${num * 100}%`;
+}
+const result = convertToPercentage(0.3);
+```
+
+Then we can print the value of the result to the terminal using `console.log`.
+
+```js {linenos=table,hl_lines=["6"],linenostart=1}
+function convertToPercentage() {
+  const num = 0.5;
+  const percentage = `${num * 100}%`;
+}
+const result = convertToPercentage(0.3);
+console.log(result);
+```
+
+If our function behaves as intended, we expect the output to be `"30%"`.
+
+{{<note type="question" title="exercise">}}
+Given the code above, try predicting what `result` will evaluate to.
+
+{{</note>}}
+
+### Undefined values
+
+When we log result's value to the terminal we get a new data type - `undefined`.  
+`undefined` is a data type used to represent the absence of a value.  
+We need to specify the value want to output from our function. If we don't state what our function should output then the default output will be `undefined`.
+
+### Setting an output
```

Comment:
It's reference the idea of expressions evaluating to values again here - it was heavily referenced in week 1, I think we should be reinforcing that terminology and mental model here.